neuraxle/base.py

```diff
@@ -3456,3 +3596,60 @@ def load(self, context: ExecutionContext, full_dump=True) -> BaseStep:
 
         context.pop()
         return loaded_self.load(context, full_dump)
+
+
+class _WithContextStepSaver(BaseSaver):
+    """
+    Custom saver for steps that have an :class:`ExecutionContext`.
+    Loading will inject the saved dependencies inside the :class`ExecutionContext`.
+    .. seealso::
+        :class:`_HasContext`,
+        :class:`BaseSaver`,
+        :class:`ExecutionContext`
+    """
+
+    def load_step(self, step: 'StepWithContext', context: ExecutionContext) -> 'StepWithContext':
+        """
+        Load a step with a context by setting the context as the loading context.
+
+        :param step: step with context
+        :param context: execution context to load from
+        :return: loaded step with context
+        """
+        step.context = context
+        step.apply('_assert_has_services', context=context)
+        return step
+
+    def save_step(self, step: 'StepWithContext', context: ExecutionContext) -> 'StepWithContext':
+        """
+        If needed, remove parents of a step with context before saving.
+
+        :param step: step with context
+        :param context: execution context to load from
+        :return: saved step with context
+        """
+        del step.context
+        return step
+
+    def can_load(self, step: 'StepWithContext', context: 'ExecutionContext'):
+        return True
+
+class StepWithContext(ForceHandleMixin, MetaStepMixin, BaseTransformer):
```

Comment:
Isnt it a base step rather than a base transformer ? I strongly think that it can fit with a context. I woudln't have thought that we'd disallow fitting here.

Reply:
Just this left and we merge @alexbrillant